addons/hr/models/hr_employee_base.py

```diff
@@ -74,7 +74,7 @@ def _compute_newly_hired(self):
         new_hire_field = self._get_new_hire_field()
         new_hire_date = fields.Datetime.now() - timedelta(days=90)
         for employee in self:
-            employee.newly_hired = employee[new_hire_field] > new_hire_date
+            employee.newly_hired = employee[new_hire_field] > new_hire_date if isinstance(employee[new_hire_field], datetime) else False
```

Comment:
We could manage the case when this is a date and not a datetime (eg when the field ``first_contract_date`` is used)
```suggestion
            if not employee[new_hire_field]:
                employee.newly_hired = False
            elif not isinstance(employee[new_hire_field], datetime):
                employee.newly_hired = employee[new_hire_field] > new_hire_date.date()
            else:
                employee.newly_hired = employee[new_hire_field] > new_hire_date
```